ql/src/java/org/apache/hadoop/hive/ql/parse/UpdateDeleteSemanticAnalyzer.java

```diff
@@ -99,14 +101,31 @@ protected void analyze(ASTNode tree, Table table, ASTNode tabNameNode) throws Se
    */
   private void reparseAndSuperAnalyze(ASTNode tree, Table mTable, ASTNode tabNameNode) throws SemanticException {
     List<? extends Node> children = tree.getChildren();
-
+    
+    boolean shouldTruncate = children.size() == 1 && deleting();
+    if (shouldTruncate) {
+      StringBuilder rewrittenQueryStr = new StringBuilder("truncate ").append(getFullTableNameForSQL(tabNameNode));
+      ReparseResult rr = parseRewrittenQuery(rewrittenQueryStr, ctx.getCmd());
+      Context rewrittenCtx = rr.rewrittenCtx;
+      ASTNode rewrittenTree = rr.rewrittenTree;
+
+      TruncateTableAnalyzer truncate = (TruncateTableAnalyzer) SemanticAnalyzerFactory.get(queryState, rewrittenTree);
```

Comment:
Does this cast required? I don't see any TruncateTableAnalyzer specific calls.

Reply:
fixed